api/Dockerfile

```diff
@@ -1,3 +1,21 @@
+# Use the ubuntu:bionic in order to run the tests
+FROM ubuntu:bionic AS build
+
+# Update packages
+RUN apt-get update 
+RUN apt-get install -y openjdk-8-jdk maven
+
+# Set the correct Java version
+RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+
+# Copy POM and source into the /tmp for building.
+COPY pom.xml /tmp/
+COPY src /tmp/src/
+
+# Set Working Directory
+WORKDIR /tmp/
+RUN mvn test package
```

Comment:
will this stop the docker build on failure?

Reply:
Yes... Maven exits with a non-zero code stopping the build as shown in the execution below:

[ERROR] Failed to execute goal ### org.apache.maven.plugins:maven-surefire-plugin:2.21.0:test (default-test) on project spring-dal: **There are test failures.**
[ERROR] 
[ERROR] Please refer to /tmp/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
The command '/bin/sh -c mvn test package' returned a non-zero code: 1